controller/api/destination/traffic_split_adaptor.go

```diff
@@ -53,7 +53,10 @@ func (tsa *trafficSplitAdaptor) publish() {
 	if tsa.profile != nil {
 		merged = *tsa.profile
 	}
-	if tsa.split != nil {
+
+	// Update only if `dstOverrides` is empty, to give preference
+	// to serviceprofiles over trafficsplits
+	if tsa.split != nil && len(merged.Spec.DstOverrides) == 0 {
```

Comment:
Asking this mainly for my own understanding, What is the concrete scenario that represents this condition. From my reading of this, it is when we have a trafficsplit for a service and the we don't have destination overrides for this service. However, further down in the code, we always set `DstOverrides` to a non-empty list.

Reply:
>  What is the concrete scenario that represents this condition. From my reading of this, it is when we have a trafficsplit for a service and the we don't have destination overrides for this service. 

We should be setting `sp.dstOverrides` to `trafficsplit.split` **only** when `dstOverrides` is empty i.e giving preference to `dstOverrides` over `trafficsplit`. Further down in the code, We set it to a non-empty list, when `dstOverrdies` is empty even then i.e both `trafficsplit` and `sp.dstOverrides` were not set/empty. Does that help?